src/simple.sml

```diff
@@ -50,6 +51,65 @@ struct
            List.find (fn (su,r) => r = r') (map #1 (List.nth (#hands s,i)))
        | _ => NONE
 
+  (* Gives the clued rank of a card, returns NONE if no number clue has been given *)
+  fun cluedRank (is : info list) : int option =
+    case List.find (fn i => case i of IsRank _ => true | _ => false) is of
+      SOME (IsRank r) => SOME r
+     | _ => NONE
+
+  (* Gives the list of negative rank information of a card *)
+  fun cluedNotRank (is : info list) : int list =
+    List.mapPartial (fn i => case i of NotRank r => SOME r | _ => NONE) is
+
+  (* Gives the list of negative color information of a card *)
+  fun cluedNotSuit (is : info list) : suit list =
+    List.mapPartial (fn i => case i of NotSuit su => SOME su | _ => NONE) is
+
+  (* Returns whether a card is clued to be not rainbow *)
+  fun isNotRainbow (is : info list) : bool =
```

Comment:
Isn't this just `not (null (cluedNotSuit is))`? I think you can just inline that inside of `possibleSuits`.